spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/route/builder/GatewayFilterSpec.java

```diff
@@ -123,14 +124,14 @@ public GatewayFilterSpec hystrix(Consumer<HystrixGatewayFilterFactory.Config> co
 		return filter(factory.apply(this.routeBuilder.getId(), configConsumer));
 	}
 
-	public <T, R> GatewayFilterSpec modifyRequestBody(Class<T> inClass, Class<R> outClass, RewriteFunction<T, R> rewriteFunction) {
+	public <T, R> GatewayFilterSpec modifyRequestBody(Class<T> inClass, Class<R> outClass, RewriteFunction<T, R> rewriteFunction, Map<String, Object> inHints, Map<String, Object> outHints) {
```

Comment:
Since RC1 is already public, let's leave the original methods and either overload, or create a spec.

Reply:
Don't worry about RC1, just keep the originals and either overload or create a spec similar to what is in your branch.

Reply:
So, I'll overload methods with specs. Is it OK? If yes, tomorrow I'll send you PR to master branch, keeping existed signatures as they were before.

OK, tomorrow I am going to review what you write here... It's late (Moscow time 10 p.m.), I'm pretty tired and I'm going to go to bed.